### PR TITLE
nitcorn: FileServer default file

### PR DESCRIPTION
**Pull request by Morriar**
This PR introduce the concept of default file in `FileServer`. This allow the user to specify what file to return instead of the 404 page if no file matches the query.

### CODE REVIEW

**Comment by Morriar** [comment]
test this please

File: lib/nitcorn/file_server.nit [hunk 1]
```diff
@@ -106,7 +106,7 @@ class FileServer
 					end
 				end
 
-				var is_dir = local_file.file_stat.is_dir
+				var is_dir = local_file.file_stat.as(not null).is_dir
```
Comment (privat) [inline lib/nitcorn/file_server.nit]: if you use a `file_exists`, then a `file_stat.as(not null)`  then just use `file_stat`

~~~nit
vat stat = local_file.file_stat
if stat != null then # was if local_file.file_exists then
  if stat.is_dir then #  was if local_file.file_stat.is_dir then
  # ...
~~~

File: lib/nitcorn/file_server.nit [hunk 2]
```diff
@@ -124,6 +130,20 @@ class FileServer
...
+		var default_file = self.default_file
+		if default_file == null then
+			return new HttpResponse(404)
+		end
+
+		var local_file = (root / default_file).simplify_path
+		if not local_file.file_exists then
+			return new HttpResponse(404)
+		end
+		return answer_file(local_file)
```
Comment (privat) [inline lib/nitcorn/file_server.nit]: `answer_file` does already the file does not exist 404 check. No need to do it here again

**Comment by privat** [comment]
+1. However I have a remark. In most web servers, returning a specific 404 page need some special care as the same web content (the file) is served by different url, therefore the ressources with a relative uri in the content will be easy to break. Maybe add a note about it in the documentation?

**Comment by Morriar** [comment]
Reroll with @privat remarks except the last one.

**Comment by privat** [comment]
test this please

**Comment by privat** [comment]
What's new in the last reroll?

**Pull request merged**